### PR TITLE
Enable nullability in ControlCollection

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in ControlCollection.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6277)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs [hunk 1]
```diff
@@ -204,7 +202,10 @@ public bool Contains(Control control)
             /// </summary>
             public Control[] Find(string key, bool searchAllChildren)
             {
-                key.ThrowIfNullOrEmptyWithMessage(SR.FindKeyMayNotBeEmptyOrNull);
+                if (string.IsNullOrEmpty(key))
+                {
+                    throw new ArgumentNullException(nameof(key), SR.FindKeyMayNotBeEmptyOrNull);
+                }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:208]:
```suggestion
                key.ThrowIfNullOrEmptyWithMessage(SR.FindKeyMayNotBeEmptyOrNull);
```

**Review by RussKie — COMMENTED** [review]
👍

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs [hunk 2]
```diff
@@ -385,7 +383,7 @@ public virtual void RemoveByKey(string key)
                             string.Format(SR.IndexOutOfRange, index.ToString(CultureInfo.CurrentCulture)));
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:383]: what about method return type here?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:383]: Sorry, what do you mean? Are you suggesting changing to `public new virtual Control? this[int index]` or are you talking about something else?

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:383]: that was it. Did you leave it out intentionally?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:383]: I noticed that there is a `Debug.Assert` call, so I thought that this should not be null.

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlCollection.cs:383]: But the control being returning is already nullable at line 386. In general use, we shouldn't be expecting a null and given that we are asserting, maybe we should go with null forgiving operator on line 386.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**